packages/types/src/packages/api-reference-configuration.ts

```diff
@@ -0,0 +1,265 @@
+   * @returns A string ID used to generate the URL hash
+   * @default (heading) => `#description/${heading.slug}`
+   */
+  generateHeadingSlug: z.function().args(z.any()).returns(z.string()).optional(),
+  /**
+   * Customize the model portion of the hash
+   * @param model - The model object with a name property
+   * @returns A string ID used to generate the URL hash
+   * @default (model) => slug(model.name)
+   */
+  generateModelSlug: z
+    .function()
+    .args(z.object({ name: z.string() }))
+    .returns(z.string())
+    .optional(),
+  /**
+   * Customize the tag portion of the hash
+   * @param tag - The tag object
+   * @returns A string ID used to generate the URL hash
+   * @default (tag) => slug(tag.name)
+   */
+  generateTagSlug: z.function().args(z.any()).returns(z.string()).optional(),
+  /**
+   * Customize the operation portion of the hash
+   * @param operation - The operation object
+   * @returns A string ID used to generate the URL hash
+   * @default (operation) => `${operation.method}${operation.path}`
+   */
+  generateOperationSlug: z
+    .function()
+    .args(
+      z.object({
+        path: z.string(),
+        operationId: z.string().optional(),
+        method: z.string(),
+        summary: z.string().optional(),
+      }),
+    )
+    .returns(z.string())
+    .optional(),
+  /**
+   * Customize the webhook portion of the hash
+   * @param webhook - The webhook object
+   * @returns A string ID used to generate the URL hash
+   * @default (webhook) => slug(webhook.name)
+   */
+  generateWebhookSlug: z
+    .function()
+    .args(
+      z.object({
+        name: z.string(),
+        method: z.string().optional(),
+      }),
+    )
+    .returns(z.string())
+    .optional(),
+  /** Callback fired when the reference is fully loaded */
+  onLoaded: z.union([z.function().returns(z.void()), z.undefined()]).optional(),
+  /** Base URL for the API server */
+  baseServerURL: z.string().optional(),
+  /** List of OpenAPI server objects */
+  servers: z.array(z.any()).optional(), // Using any for OpenAPIV3_1.ServerObject
+  /** Whether to include default fonts */
+  withDefaultFonts: z.boolean().optional(),
+  /** Whether to expand all tags by default */
+  defaultOpenAllTags: z.boolean().optional(),
+  /**
+   * Function to sort tags
+   * @default 'alpha' for alphabetical sorting
+   */
+  tagsSorter: z.union([z.literal('alpha'), z.function().args(z.any(), z.any()).returns(z.number())]).optional(),
+  /**
+   * Function to sort operations
+   * @default 'alpha' for alphabetical sorting
+   */
+  operationsSorter: z
+    .union([z.literal('alpha'), z.literal('method'), z.function().args(z.any(), z.any()).returns(z.number())])
+    .optional(),
+  /** Integration type identifier */
+  _integration: IntegrationEnum.optional(),
+  /** Whether to hide the client button */
+  hideClientButton: z.boolean().optional(),
+})
+
+export type ApiReferenceConfiguration = z.infer<typeof ApiReferenceConfigurationSchema>
```

Comment:
Also we should have two types here.
z.input is for the user, it's before zod validation.

z.infer is for us. It's after zod validation